### ISSUE TITLE
False positives when using macros in metadata test function

### ISSUE THREAD

**Issue by borkdude** [borkdude/clj-kondo#597]
``` clojure
(ns repro
  (:require [cljs.test :refer [is testing]]))


(defn pretty-print
  {:test (fn []
           (testing "Default printing options"
             (let [res (as-> (repeat 1 "foo") x)]
               (is res))))})
```

```
$ clj-kondo --lint repro.cljs
/tmp/pprint.cljs:8:20: error: unresolved symbol res
/tmp/pprint.cljs:8:47: error: unresolved symbol x
linting took 14ms, errors: 2, warnings: 0
```
